include/yamail/resource_pool/error.hpp

```diff
@@ -30,11 +30,11 @@ enum code {
 
 namespace detail {
 
-class category : public boost::system::error_category {
+class category final : public boost::system::error_category {
 public:
-    const char* name() const noexcept { return "yamail::resource_pool::error::detail::category"; }
+    const char* name() const noexcept override final { return "yamail::resource_pool::error::detail::category"; }
 
-    std::string message(int value) const {
+    std::string message(int value) const override final {
```

Comment:
Just final is enough: https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#c128-virtual-functions-should-specify-exactly-one-of-virtual-override-or-final

Reply:
You are welcome to edit the PR to reflect that, but I disagree with the core guidelines document, so don't wish to do so.

My objective is to silence the warning. I don't mind if you close this PR and change the code the way you want.

Reply:
Probably I will this on my own. But I’m still interested in arguments against this guide. For me it seems useful because it’s better to have one keyword instead of two to do exactly the same thing. Just less to read.

Reply:
I have had numerous co-workers submit pull requests to my code to add the override keyword to functions already marked as final. So from a practical perspective, this avoids people submitting pull requests to add 'override' in the future.

My objection is that the language allows both override, and final, and virtual, to be placed on a function, and then the standards committee publishes a non-standard document (The Core Guidelines) telling people that they're using the language incorrectly. If the intention was really to only allow zero or one of those keywords on any specific function, that rule should have been encoded into the C++ standard directly.

Further, I have a small recollection (Though, to be honest, I don't know how accurate this is) of older MSVC compilers triggering a warning on functions that were marked final, but not override. Again, not sure how accurate this memory is.

Reply:
Thanks for the arguments.